src/core/service.c

```diff
@@ -3330,6 +3330,12 @@ static void service_notify_cgroup_empty_event(Unit *u) {
 
                 break;
 
+        /* If the cgroup empty notification comes when the unit is not active, we must have failed to clean
+         * up the cgroup earlier and should do it now. */
+        case SERVICE_DEAD:
+        case SERVICE_FAILED:
+                unit_prune_cgroup(u);
```

Comment:
should really have a "break" here. Or  a `_fallthrough_`, but that would be weird